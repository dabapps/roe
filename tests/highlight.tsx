import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {
  Column,
  Highlight,
  InputGroup,
  InputGroupAddon,
  Row,
} from '../src/ts/';

describe('Highlight', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(
      <Highlight>
        <p>Hello, World!</p>
      </Highlight>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with props (open)', () => {
    const tree = renderer.create(
      <Highlight open>
        <p>Hello, World!</p>
      </Highlight>
    );

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with props (open, disabled, backgroundColor)', () => {
    const tree = renderer.create(
      <Highlight open disabled backgroundColor="white">
        <p>Hello, World!</p>
      </Highlight>
    );

    expect(tree).toMatchSnapshot();
  });
});
