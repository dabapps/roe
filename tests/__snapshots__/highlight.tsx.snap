// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should match snapshot 1`] = `
<div
  className="highlight"
>
  <div />
  <div
    className="highlight-content"
    style={undefined}
  >
    <p>
      Hello, World!
    </p>
  </div>
</div>
`;

exports[`Highlight should match snapshot with props (open) 1`] = `
<div
  className="highlight"
>
  <div>
    <div
      className="highlight-overlay"
    />
  </div>
  <div
    className="highlight-content open"
    style={undefined}
  >
    <p>
      Hello, World!
    </p>
  </div>
</div>
`;

exports[`Highlight should match snapshot with props (open, disabled, backgroundColor) 1`] = `
<div
  className="highlight"
>
  <div>
    <div
      className="highlight-overlay"
    />
  </div>
  <div
    className="highlight-content open"
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <p>
      Hello, World!
    </p>
    <div
      className="highlight-overlay-disabled"
    />
  </div>
</div>
`;
